libvirt/resource_libvirt_domain.go

```diff
@@ -629,6 +633,20 @@ func resourceLibvirtDomainUpdate(d *schema.ResourceData, meta interface{}) error
 		d.SetPartial("cloudinit")
 	}
 
+	if d.HasChange("tags") {
+		out := TerraformInstanceXML{}
```

Comment:
This should also not be empty. domain.GetMetadata() should grab the existing information and merge the tags on top